lib/reverse_proxy_plug.ex

```diff
@@ -178,49 +178,47 @@ defmodule ReverseProxyPlug do
     |> Conn.resp(status, body)
   end
 
-  if {:module, HTTPoison} == Code.ensure_loaded(HTTPoison) do
-    # This section of the code is present for retrocompatibility
-    # for the case where HTTPoison is the underlying HTTP Client
-    defp process_response(:stream, conn, _resp, opts),
-      do: stream_response(conn, opts)
-
-    @spec stream_response(Conn.t(), Keyword.t()) :: Conn.t()
-    defp stream_response(conn, opts) do
-      receive do
-        %HTTPoison.AsyncStatus{code: code} ->
-          case opts[:status_callbacks][code] do
-            nil ->
-              conn
-              |> Conn.put_status(code)
-              |> stream_response(opts)
-
-            handler ->
-              handler.(conn, opts)
-          end
-
-        %HTTPoison.AsyncHeaders{headers: headers} ->
-          headers
-          |> normalize_headers
-          |> Enum.reject(fn {header, _} -> header == "content-length" end)
-          |> Enum.concat([{"transfer-encoding", "chunked"}])
-          |> Enum.reduce(conn, fn {header, value}, conn ->
-            Conn.put_resp_header(conn, header, value)
-          end)
-          |> Conn.send_chunked(conn.status)
-          |> stream_response(opts)
-
-        %HTTPoison.AsyncChunk{chunk: chunk} ->
-          case Conn.chunk(conn, chunk) do
-            {:ok, conn} ->
-              stream_response(conn, opts)
-
-            {:error, :closed} ->
-              conn
-          end
-
-        %HTTPoison.AsyncEnd{} ->
-          conn
-      end
+  # This section of the code is present for retrocompatibility
```

Comment:
This was removed because HTTPoison has been made explicitly mandatory.
IMO the preferred fix would be to make it explicitly optional, with the proper warning in the readme, but this would warrant a bump to 2.0.0 due to retrocompatibility issues, instead of 1.5.0/1.5.1

Reply:
@polvalente is it necessary to make HTTPoison mandatory? Is there a reason that an approach like #141 does not work?

Reply:
@axelson it isn't necessary, but it would mean a breaking change. It all depends on the direction the lib wants to go :)

Reply:
Ah, I see. That makes sense :+1: